Java-Data-Types.md

```diff
@@ -0,0 +1,148 @@
+# Java data-Types
+
+Java is a strongly ­typed language. This means that, in Java, each data type has its own strict definition. There are no implicit data type conversions when any conflicts occur between the data types. Any change in data types should be explicitly declared by the programmer.
+ 
+Java defines 8 simple data types. They are divided into the following categories:
+ 
+Integers
+ 
+Floating­ Point Numbers
+ 
+Characters
+ 
+Boolean Type
+ 
+The details of each of the data types is given below
+ 
+### Integers:
+
+These are of four types: byte, short, int, long. It is important to note that these are signed positive and negative values. Java does not support positive­ only numbers.
+
+#### byte:
+
+    Byte data type is an 8-bit signed two's complement integer.
+
+    Minimum value: -128 (-2^7)
+
+    Maximum value: 127 (inclusive)(2^7 -1)
+
+    Default value: 0
+
+    Example: byte a = 10 , byte b = -500
+
+#### short:
+
+    Short data type is a 16-bit signed two's complement integer.
+
+    Minimum value: -32,768 (-2^15)
+
+    Maximum value: 32,767 (inclusive) (2^15 -1)
+
+    Default value: 0.
+
+    Example: short s = 10, short r = -1000
+
+#### int:
+
+    Int data type is a 32-bit signed two's complement integer. Int is generally used as the default data type 
+    for integral values unless there is a concern about memory.
+
+    Minimum value: (-2^31)
+
+    Maximum value: (2^31 -1)
+
+    The default value: 0.
+
+    Example: int a = 50000, int b = -20
+
+#### long:
+
+    Long data type is a 64-bit signed two's complement integer.
+
+    Minimum value: (-2^63)
+
+    Maximum value: (2^63 -1)
+
+    Default value: 0L.
+
+    Example: long a = 100000L, long b = -600000L
+
+
+### Floating­ Point​:
+
+These are also called real numbers and are used for expressions involving fractional precision. These are of two types: float, double. Float is actually avoided in case of precise data such as currency or research data.
+
+#### float:
+
+    Float data type is a single-precision 32-bit IEEE 754 floating point.
+
+    Float is mainly used to save memory in large arrays of floating point numbers.
+
+    Default value: 0.0f.
```

Comment:
Same as above. You are using `f` at the end of the number literal, to direct the compiler to store it as floating point. Mention that clearly.

Also note that if you try to do `float myFloat = 7.8`, it would give a compiler error. A lot of beginners get put off by this.